spec/controllers/api/v1/layouts_controller_spec.rb

```diff
@@ -6,34 +6,57 @@
   def validate_layout(layout_json, layout)
     expect(layout_json[:id]).to eq(layout.id.to_s)
     expect(layout_json[:attributes][:name]).to eq(layout.name)
+
+    expected_locations_json = layout_json[:relationships][:locations][:data]
+    expect(expected_locations_json.size).to eq(layout.locations.size)
+    expected_locations_json.zip(layout.locations).each do |json_location, orig_location|
+      expect(json_location[:id]).to eq(orig_location.id.to_s)
+    end
+  end
+
+  def validate_included(locations_json, locations)
+    locations_json.zip(locations).each do |json_location, orig_location|
+      expect(json_location[:id]).to eq(orig_location.id.to_s)
+      expect(json_location[:attributes][:name]).to eq(orig_location.name)
+    end
   end
 
   describe "GET #show" do
     it "should return a serialized layout instance" do
-      layout = create(:layout)
+      layout = create(:layout_with_locations)
```

Comment:
I would also change the standard :layout factory to have locations, seeing as it should be invalid without them.